lib/Runtime/PlatformAgnostic/Intl.h

```diff
@@ -77,41 +77,81 @@ namespace Intl
         void Cleanup() override
         {
             delete intlObject;
-            intlObject = nullptr;
+            delete this;
```

Comment:
>delete this; [](start = 12, length = 12)

is this actually something one can do? It seems like deleting this before returning from this method might be undefined behavior (although the method isn't tied to the lifetime of the object this)...

A bit of reading tells me that `delete this` calls the destructor of this class. Are we happy with what the destructor does in this case?

Reply:
Its the default destructor, so as long as it free's, we are happy. It is @MSLaguana's suggestion, I thought it was fairly scary too.